src/state/opamSysInteract.ml

```diff
@@ -734,6 +839,7 @@ let update ?(env=OpamVariable.Map.empty) () =
     | Gentoo -> Some (`AsAdmin "emerge", ["--sync"])
     | Homebrew -> Some (`AsUser "brew", ["update"])
     | Macports -> Some (`AsAdmin "port", ["sync"])
+    | Nixos -> Some (`AsUser "nix-channel", ["--update"])
```

Comment:
This won't do anything if the NixOS system is (like mine) using flakes to provide a instance Nixpkgs instances instead of channels. However, we definitely shouldn't call `nix flake lock --update-input nixpkgs` from opam so I don't have a better suggestion.

A more general point: does this have opam invoking a command to update the entire system? I notice `pacman` is invoked with `-Sy`, which is [not recommended](https://wiki.archlinux.org/title/System_maintenance#Avoid_certain_pacman_commands) as refreshing wityhout updating results in partial upgrades. I think this should be `-Syu`. Unless there's a good reason not too I'll create a new issue for this.

Reply:
This is mostly use for CI systems that really benefit from a non-platform way of updating the system package repository.

> A more general point: does this have opam invoking a command to update the entire system? I notice `pacman` is invoked with `-Sy`, which is [not recommended](https://wiki.archlinux.org/title/System_maintenance#Avoid_certain_pacman_commands) as refreshing wityhout updating results in partial upgrades. I think this should be `-Syu`. Unless there's a good reason not too I'll create a new issue for this.

From opam's point of view this is safe to do because installing a package is done using `pacman -Su`.

Reply:
> This won't do anything if the NixOS system is (like mine) using flakes to provide a instance Nixpkgs instances instead of channels.

I feel like I'm always learning new ways to use nix.